dependencies.gradle

```diff
@@ -5,6 +5,13 @@ ext {
   kotlinVersion = project.hasProperty('square.kotlinVersion') ?
       project.getProperty('square.kotlinVersion') : '1.3.72'
 
+  kotlinUseIR = (project.hasProperty('square.useIR') ?
+      project.getProperty('square.useIR') : 'false').toBoolean()
+
+  if (kotlinUseIR && !kotlinVersion.startsWith('1.4.2')) {
+    throw new GradleException("The new IR backend is only testable with Kotlin 1.4.20 or newer.")
```

Comment:
A bit of a mismatch here – you're only allowing this on `1.4.20` _or newer_, but only checking `startsWith('1.4.2')`. This would break if you were, for example, running against kotlin version `1.4.30`.

Should you be doing some kind of semver-aware check here, comparing version codes with something like `org.gradle.util.VersionNumber`?

Reply:
Added `kotlinIrSupported = VersionNumber.parse(kotlinVersion).baseVersion >= VersionNumber.parse("1.4.20")`.

Reply:
Nice!